### PR TITLE
CLI: Updates `tkn hub info task` cli command for --version flag to include deprecation info

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes

    
- This patch adds deprecation info along with version
   for a deprecated version of a resource
    
    Signed-off-by: Shiv Verma <shverma@redhat.com>

Screenshot from 2021-07-19 16-29-40

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/pkg/cli/formatter/field.go [hunk 1]
```diff
@@ -124,10 +124,13 @@ func findSpaceIndexFromLast(str string) int {
 
 // FormatVersion returns version appended with (latest) if the
 // latest field passed is true
-func FormatVersion(version string, latest bool) string {
+func FormatVersion(version string, latest bool, deprecated bool) string {
 	if latest {
 		return version + " (Latest)"
 	}
+	if deprecated {
```
Comment (piyush-garg) [inline api/pkg/cli/formatter/field.go:131]: Handle the case if latest version is deprecated?

Comment (pratap0007) [reply api/pkg/cli/formatter/field.go:131]: yes

File: api/pkg/cli/cmd/info/info.go [hunk 2]
```diff
@@ -194,11 +195,19 @@ func (opts *options) run() error {
 	out := opts.cli.Stream().Out
 
 	resVersion := resource.(hub.ResourceWithVersionData)
+
+	var deprecated bool
+	if resVersion.Deprecated != nil {
```
Comment (piyush-garg) [inline api/pkg/cli/cmd/info/info.go:200]: cant we use bool now

Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/info/info.go:200]: +1

Comment (pratap0007) [reply api/pkg/cli/cmd/info/info.go:200]: we can't use ,In response for non deprecated version of a resource doesn't contains deprecated filed so needs to check it

**Pull request merged**